### DISCUSSION TITLE
UserWarning: [W030] Some entities could not be aligned in the text

### DISCUSSION THREAD

**Discussion by davesuketu215** [explosion/spaCy#6979] [Help: Coding & Implementations] [answered]
While training my custom spacy model on my training data, I receive the below warning messages:
```
UserWarning: [W030] Some entities could not be aligned in the text "Confirmation with wrong net value <p>. </p> For sh..." with entities "[(1731, 1734, 'ORG'), (2054, 2057, 'ORG'), (1167, ...". Use `spacy.training.offsets_to_biluo_tags(nlp.make_doc(text), entities)` to check the alignment. Misaligned entities ('-') will be ignored during training.
  entities=ent_str[:50] + "..." if len(ent_str) > 50 else ent_str,
```

Is there any way in which I can avoid these errors?

Btw, I am annotating my dataset in this format: ("Text", {"entities": [(41, 53, "ORG")]})

**Comment by adrianeboyd** [comment, marked as answer]
Hi, this means that some of your entity character spans don't align with token boundaries for the tokenizer you're training with. You can check the character spans with `doc.char_span`:

```python
doc = nlp.make_doc(text)
span = doc.char_span(start_char, end_char, label=label)
```

If the returned `span` is `None`, this means that the token boundaries didn't line up.

It if makes sense for your data, you could automatically adjust the start/end by using `doc.char_span(alignment_mode=)` to adjust the boundaries to the closest token boundaries inside or outside the character span. For example:

```python
span = doc.char_span(start_char, end_char, label=label, alignment_mode="expand")
adjusted_start_char = span.start_char
adjusted_end_char = span.end_char
```

In a lot cases, you may need to manually review the annotation, since changing the character span could change or invalidate the entity type and potentially lead to noisy training data.

**Reply by davesuketu215** [reply]
Upon ignoring these errors, the implication would be loss of training data, right?

**Reply by svlandeg** [reply]
That's right - misaligned entities  will be ignored during training.

**Reply by adrianeboyd** [reply]
Yes, it will ignore these tokens during training (as opposed to treating them as `O`, which would be confusing for the model).
